pkg/processor/trigger/partitioned/v3io/partition.go

```diff
@@ -118,3 +95,39 @@ func (p *partition) Read() error {
 		}
 	}
 }
+
+func (p *partition) seek(partitionPath string) (string, error) {
+	p.Logger.DebugWith("Seeking partition",
+		"partitionPath", partitionPath,
+		"seekType", p.v3ioTrigger.seekType)
+
+	response, err := p.v3ioTrigger.container.Sync.SeekShard(&v3iohttp.SeekShardInput{
+		Path: partitionPath,
+		Type: p.v3ioTrigger.seekType,
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "Failed to seek partition")
+	}
+	defer response.Release()
+
+	location := response.Output.(*v3iohttp.SeekShardOutput).Location
+	if location == "" {
+		return "", errors.New("Got empty location from seek")
+	}
+	return location, nil
+}
+
+func (p *partition) getRecords(partitionPath string, location string, pollingInterval time.Duration) (*v3iohttp.GetRecordsOutput, error) {
+	response, err := p.v3ioTrigger.container.Sync.GetRecords(&v3iohttp.GetRecordsInput{
+		Path:     partitionPath,
+		Location: location,
+		Limit:    p.v3ioTrigger.configuration.ReadBatchSize,
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer response.Release()
```

Comment:
if there's an error the defer won't be called, so why use defer?

Reply:
Just in case the below panics on the cast.

Reply:
I guess actually this can be done immediately. I mean, `response.Output` is already populated.